Tests/TuistCacheTests/ContentHashing/ContentHasherTests.swift

```diff
@@ -0,0 +1,108 @@
+import TSCBasic
+import TuistSupport
+import XCTest
+@testable import TuistCache
+@testable import TuistSupportTesting
+
+final class ContentHasherTests: TuistUnitTestCase {
+    private var subject: ContentHasher!
+    private var mockFileHandler: MockFileHandling!
+
+    override func setUp() {
+        super.setUp()
+        mockFileHandler = MockFileHandling()
+        subject = ContentHasher(fileHandler: mockFileHandler)
+    }
+
+    override func tearDown() {
+        subject = nil
+        mockFileHandler = nil
+        super.tearDown()
+    }
+
+    // MARK: - Tests
+
+    func test_hashstring_foo_returnsItsMd5() throws {
+        // Given
+        let hash = try subject.hash("foo")
+
+        // Then
+        XCTAssertEqual(hash, "acbd18db4cc2f85cedef654fccc4a4d8")
+    }
+
+    func test_hashstring_bar_returnsItsMd5() throws {
+        // Given
+        let hash = try subject.hash("bar")
+
+        // Then
+        XCTAssertEqual(hash, "37b51d194a7513e45b56f6524f2d51f2")
+    }
+
+    func test_hashstrings_foo_bar_returnsAnotherMd5() throws {
+        // Given
+        let hash = try subject.hash(["foo", "bar"])
+
+        // Then
+        XCTAssertEqual(hash, "3858f62230ac3c915f300c664312c63f")
+    }
+
+    func test_hashFile_hashesTheExpectedFile() throws {
+        // Given
+        let data = Data(repeating: 1, count: 10)
```

Comment:
Without your mock file handler, here you'd create a file in a temporary directory, and then get the checksum of that file from the disk. As I mentioned in another comment, that makes the test a bit slower, but the scenario more realistic, so I think it's a safe tradeoff to make.